### ISSUE TITLE
Javascript-obfuscator together with angular-cli

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [javascript-obfuscator/javascript-obfuscator#342]
Hey guys,

I have a big Angular app with multiple lazy loading chunks I would like to obfuscate while building for production.
I'm working with angular-cli 6 (soon to be 7) and would like to add it in my release flow so that it runs automatically each time I release to production.
Preferably, without having to eject the angular-cli configuration.
Is this possible? If so, is there a sort of tutorial on how to do this?

thanks for the library!

Best regards,

Lander

**Comment by markuszm** [comment]
While there is a webpack plugin of this obfuscator as you said you would need to eject the webpack config at the moment as angular cli does not support webpack plugins.

What I did was use the [gulp plugin](https://github.com/javascript-obfuscator/gulp-javascript-obfuscator) or you can also use the [grunt plugin](https://github.com/javascript-obfuscator/grunt-contrib-obfuscator) if you prefer that. Then you can run the `ng build` command and afterwards run gulp.

If you use sourcemaps you need to combine the ones that angular cli outputs with the ones that are generated from this obfuscator. There you can use https://github.com/azu/multi-stage-sourcemap. 

This obfuscator works relatively well with angular code but sometimes there can be incompatibilities with uglify which is used to minimize code in angular-cli and this obfuscator so I would recommend you to always test the obfuscated code before you deploy it.

**Comment by stephengardner** [comment]
@markuszm can you share your config of Gulp?  Specifically the combination of the sourcemaps.  I'm trying to set this up and that would be helpful

**Comment by markuszm** [comment]
Here is our gulpfile.js.
https://gist.github.com/markuszm/a76ec4ed2d8cf912b504b731dd60f996 
At the moment, there is no plugin for the multi-stage-sourcemap which is why we created a custom function for that which combines the sourcemap from angular with the one from the obfuscator. 
We deactivated some obfuscator options to prevent breaking the angular application so you might need to set some options different.
